File: fc-chat/demo/part3/ch3/03-1-message-system-add-channel/src/main/java/net/prostars/messagesystem/constant/ResultType.java
package net.prostars.messagesystem.constant;

public enum ResultType {
  SUCCESS("Success."),
  FAILED("failed."),
  INVALID_ARGS("Invalid arguments."),
  NOT_FOUND("Not found."),
  ALREADY_JOINED("Already joined."),
  OVER_LIMIT("Over limit."),
  NOT_JOINED("Not joined."),
  NOT_ALLOWED("Unconnected users included.");

  private final String message;

  ResultType(String message) {
    this.message = message;
  }

  public String getMessage() {
    return message;
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/constant/ResultType.kt
package com.yscorp.withpush.messagesystem.constant

enum class ResultType(val message: String) {
    SUCCESS("Success."),
    FAILED("failed."),
    INVALID_ARGS("Invalid arguments."),
    NOT_FOUND("Not found."),
    ALREADY_JOINED("Already joined."),
    OVER_LIMIT("Over limit."),
    NOT_JOINED("Not joined."),
    NOT_ALLOWED("Unconnected users included.")
}
